internal/agent/device/spec/queue.go

```diff
@@ -129,6 +129,12 @@ func (m *queueManager) Next(ctx context.Context) (*v1alpha1.Device, bool) {
 		m.log.Debugf("Template version policy updated: %d", item.Version)
 	}
 
+	if !requeue.downloadPolicySatisfied && !requeue.updatePolicySatisfied {
+		m.queue.Add(item)
+		m.log.Debugf("Template version %d policies are not satisfied skipping...", item.Version)
+		return nil, false
+	}
```

Comment:
note: this is an optimization; before, we would fail policy and rollback which is unnecessary.

Reply:
this was actually a bad idea we need the regular flow for proper status reporting and ensuring new template versions properly update the policy manager.